### ISSUE TITLE
Config threads: false freezes my tests

### ISSUE THREAD

**Issue by PavelNartov** [vitest-dev/vitest#3740]
### Describe the bug

I want my integration tests running without threads.

Then I've setup `threads: false`

And run my tests. Then it stuck on `DEV  v0.32.2` line and nothing happened.

When I remove this option or set it to true, then everything works fine, but I cannot use threads for integration tests.

I have CPU: (10) arm64 Apple M1 Pro

### Reproduction
https://github.com/PavelNartov/vitest-no-threads-stuck

### System Info

```shell
System:
    OS: macOS 12.6
    CPU: (10) arm64 Apple M1 Pro
    Memory: 76.69 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v16.13.1/bin/yarn
    npm: 8.3.0 - ~/.nvm/versions/node/v16.13.1/bin/npm
  Browsers:
    Chrome: 114.0.5735.198
    Safari: 16.0
  npmPackages:
    @vitest/ui: ^0.32.2 => 0.32.2
    vitest: ^0.32.2 => 0.32.2
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by PavelNartov** [comment]
I've changed my message and added a link to the Github repo on the Reproduction section to:
https://github.com/PavelNartov/vitest-no-threads-stuck

**Comment by PavelNartov** [comment]
Ok, I think I found an issue.

It's with a version of node. It wouldn't work on 16.13.1

But it starts to work with `node 18.16.1`

**Comment by AriPerkkio** [comment]
I can reproduce this with `16.13.1` and `16.13.2`. This seems to be fixed in NodeJS `16.14.0`.
